test/e2e/mcd_test.go

```diff
@@ -396,12 +396,12 @@ func TestReconcileAfterBadMC(t *testing.T) {
 		if err != nil {
 			return false, err
 		}
-		if mcp.Status.UnavailableMachineCount >= 1 {
+		if mcv1.IsMachineConfigPoolConditionFalse(mcp.Status.Conditions, mcv1.MachineConfigPoolDegraded) && mcp.Status.DegradedMachineCount == 1 {
```

Comment:
tests are now running with 3 maxunavailable for the worker pool so I believe `== 1` should remain `>=1` here right? or we might race?